packages/sync-actions/src/utils/create-map-action-custom.js

```diff
@@ -1,40 +1,60 @@
 import * as diffpatcher from './diffpatcher'
 
-export default function actionsMapCustom(diff, oldObj, newObj) {
+const Actions = {
+  setCustomType: 'setCustomType',
+  setCustomField: 'setCustomField',
+}
+
+const hasCustomFields = diff => Array.isArray(diff.custom)
+const hasCustomType = diff => Boolean(diff.custom.type)
+const extractCustomType = (diff, previousObject) =>
+  Array.isArray(diff.custom.type)
+    ? diffpatcher.getDeltaValue(diff.custom.type, previousObject)
+    : diff.custom.type
+const extractTypeId = (type, nextObject) =>
+  Array.isArray(type.id)
+    ? diffpatcher.getDeltaValue(type.id)
+    : nextObject.custom.type.id
+const extractTypeFields = (diffedFields, nextFields) =>
+  Array.isArray(diffedFields)
+    ? diffpatcher.getDeltaValue(diffedFields)
+    : nextFields
+const extractFieldValue = (diffedFields, nextFields, fieldName) =>
+  Array.isArray(diffedFields[fieldName])
+    ? diffpatcher.getDeltaValue(diffedFields[fieldName])
+    : nextFields[fieldName]
+
+export default function actionsMapCustom(diff, previousObject, nextObject) {
   const actions = []
   if (!diff.custom) return actions
 
-  if (Array.isArray(diff.custom)) {
+  if (hasCustomFields(diff)) {
```

Comment:
Just thinking about what these three conditions are actually trying to find out:
- `hasSingleCustomFieldChanged`
- `hasCustomTypeChanged`
- `haveMultipleCustomFieldsChanged`

Reply:
Very nice. This is what I was aiming for. Will change!